### PR TITLE
Release notes 3.29.1. Release blog and javadoc TBD

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context

I crafter release notes for 3.29.1 to enable users with our new OData client ASAP. Please, review and let's get it through.
Things to come:
- Release Blog
- Javadocs

### CODE REVIEW

File: docs/java/release-notes.mdx [hunk 1]
```diff
@@ -27,21 +27,78 @@ We highly recommend regularly updating to the latest SDK version. It will help y
...
+- We start deprecating APIs of the outdated **OData v2 client**. In this release we deprecated:
+  - Methods: `execute`, `toQuery`, `withErrorHandler`, `cachingMetadata`, `withoutCachingMetadata`, `ignoringVersionIdentifier` and `ignoreVersionIdentifier`.
+  - Error handler: `ODataVdmErrorResultHandler`.
+  - Exception classes `VersionIdentifierMissingException` and `VersionIdentifierExpiredException`.
+  - You'll find instructions how to adjust the code to use the new **OData v2 client** in the Javadoc of deprecated method or class.
+
+### New Functionality
+
+- The improved OData v2 type-safe client is now Generally Available. From now on we recommend it for OData API consumption.
+Refer to the [migration guide](./features/odata/use-typed-odata-v2-client-in-sap-cloud-sdk-for-java#switch-to-improved-odata-v2-type-safe-client) for more details on the new OData 2 client.
```
Comment (sacnl) [inline docs/java/release-notes.mdx:46]: Please, don't call it migration - it's a simple switch 😊

Comment (artemkovalyov) [reply docs/java/release-notes.mdx:46]: We don't call it migration anymore, @sacnl. :)

Comment (Emdee89) [reply docs/java/release-notes.mdx:46]: It reads "migration guide" in line 46 @artemkovalyov

Comment (artemkovalyov) [reply docs/java/release-notes.mdx]: @Emdee89 , yeah, the push has failed because of the upstream merge. Take a look at it now.

**Review by Emdee89 — APPROVED** [review]

File: docs/java/release-notes.mdx [hunk 2]
```diff
@@ -27,21 +27,78 @@ We highly recommend regularly updating to the latest SDK version. It will help y
 - update client libraries giving access to latest SAP services on SAP Cloud Platform and S4/HANA
 - protect yourself from bugs and breaking changes in the future
 
+## 3.29.1 - September 24, 2020
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.29.1)
```
Comment (jordan40) [inline docs/java/release-notes.mdx:32]: [Release blog](https://blogs.sap.com/?p=1189923)

We're still missing the Javadocs link

**Pull request merged**